### ISSUE TITLE
Doc: config meta properties

### ISSUE THREAD

**Issue by peteruithoven** [systemjs/systemjs#672]
The [Configuration API's meta section](https://github.com/systemjs/systemjs/blob/master/docs/config-api.md#meta) support properties like `format`, `exports` and `deps`. But unlike the [packages docs](https://github.com/systemjs/systemjs/blob/master/docs/config-api.md#packages) there is no elaboration on the possible properties and their functionality. 

Could someone help me by confirming, correcting the following statements? 

In the added example meta configuration is added for `vendor/angular.js`. 
**format**
Overrides in what format the module is loaded.
In the example `global` is used. 
Probably these [Module formats](https://github.com/systemjs/systemjs/blob/master/docs/module-formats.md) are allowed? 
**exports**
the global property to take as the module value
In the example `angular` is used. This means that when the meta is used outside of a packages configuration, that inside every module loaded by SystemJS an `angular` variable is available which references to the `vendor/angular.js` module? 
When the meta is added inside a packages configuration it's only available for modules inside that package?
**deps**
Dependencies to load before this module.
These dependencies are normalized with `map` and `path` configuration. 
This is currently not supported for `esm` (ES modules) (`es6`), see: https://github.com/systemjs/systemjs/issues/640.
**more options?**

**Comment by peteruithoven** [comment]
I found a lot of relevant documentation in the Module formats globals section:
https://github.com/systemjs/systemjs/blob/master/docs/module-formats.md#globals

**Comment by guybedford** [comment]
I did initially have a `docs/meta.md` that listed the meta API, but actually removed it as I found it was better to discover the options through the normal documentation as it duplicated everything. Perhaps this was a bad idea though, and a simple linking reference could still be useful.

> Probably these Module formats are allowed?

Yes exactly

> In the example angular is used. This means that when the meta is used outside of a packages configuration, that inside every module loaded by SystemJS an angular variable is available which references to the vendor/angular.js module?

No, it means that the module value of `vendor/angular.js` is taken to be the value of the `angular` global at the time of execution of the `vendor/angular.js` module. That module value is then the module value any other module will get when requesting this module as a dependency.

> When the meta is added inside a packages configuration it's only available for modules inside that package?

Yes, it's a convenience to avoid having to rewrite the package path everytime you want to add meta to a module inside a folder.

> These dependencies are normalized with map and path configuration.

Goes through full normalization cycle of `System.normalize(dep, parent)`.

> This is currently not supported for esm (ES modules) (es6), see: #640.

Yes exactly

> more options?

* `loader` for plugins
* `sourceMap` for plugin transpilers to set the source map of their transpilation
* `alias` an undocumented feature currently, which may become part of the API in future. It's basically like map config, but creates a separate module allowing for things like `deps` to be added to a mapping (as you were actually trying to do previously, and it was for this reason).

Then there are other private meta options which are not part of the public API to use, and probably best not to document.
